lib/Conversion/TorchToTosa/TorchToTosa.cpp

```diff
@@ -245,40 +251,92 @@ class ConvertAtenAddSubOp : public OpConversionPattern<AtenOpT> {
           op, "Only floating-point or integer datatype legalization supported");
     }
 
+    Type rhsAlphaMulElemType;
+    if (outElemTy.isa<mlir::FloatType>()) {
+      rhsAlphaMulElemType = outElemTy;
+    } else {
+      // if output type is 64, input type should also be 64
+      rhsAlphaMulElemType = rewriter.getIntegerType(32);
```

Comment:
This comment doesn't make sense in this context. Perhaps a comment at the top of the function describing what you are trying to achieve for i64 would be helpful.

Reply:
Done